metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvesterImpl.java

```diff
@@ -226,4 +216,32 @@ public void close() throws IOException {
       this.oaiClient.close();
     }
   }
+
+  private static class ReportingIterationWrapper implements ReportingIteration<Header> {
+
+    final ReportingIteration<OaiRecordHeader> action;
+    final Predicate<OaiRecordHeader> filter;
+
+    public ReportingIterationWrapper(ReportingIteration<OaiRecordHeader> action,
+            Predicate<OaiRecordHeader> filter) {
+      this.action = action;
+      this.filter = filter;
+    }
+
+    @Override
+    public IterationResult process(Header input) {
+      final OaiRecordHeader header = new OaiRecordHeader(input.getIdentifier(), input.isDeleted(),
+              Optional.ofNullable(input.getDatestamp()).map(Date::toInstant).orElse(null));
+      final IterationResult result;
```

Comment:
This if else can be simplified by setting a value during initialization:
e.g. IterationResult result = IterationResult.CONTINUE;

Reply:
I simplified it.